### PR TITLE
NDRS-507: Rename `global_state_hash` with `state_root_hash`.

### PR DESCRIPTION
**Pull request by goral09**
https://casperlabs.atlassian.net/browse/NDRS-507

### CODE REVIEW

File: client/src/common.rs [hunk 1]
```diff
@@ -99,14 +99,14 @@ pub mod force {
     }
 }
 
-/// Handles providing the arg for and retrieval of the global state hash.
-pub mod global_state_hash {
+/// Handles providing the arg for and retrieval of the root state hash.
```
Comment (Fraser999) [inline client/src/common.rs:102]:
```suggestion
/// Handles providing the arg for and retrieval of the state root hash.
```

File: execution_engine/src/core/engine_state/genesis.rs [hunk 2]
```diff
@@ -57,7 +57,10 @@ impl GenesisResult {
             CommitResult::KeyNotFound(key) => GenesisResult::KeyNotFound(key),
             CommitResult::TypeMismatch(type_mismatch) => GenesisResult::TypeMismatch(type_mismatch),
             CommitResult::Serialization(error) => GenesisResult::Serialization(error),
-            CommitResult::Success { state_root, .. } => GenesisResult::Success {
+            CommitResult::Success {
+                state_root_hash: state_root,
+                ..
+            } => GenesisResult::Success {
                 post_state_hash: state_root,
```
Comment (Fraser999) [inline execution_engine/src/core/engine_state/genesis.rs:64]:
```suggestion
            CommitResult::Success {
                state_root_hash,
                ..
            } => GenesisResult::Success {
                post_state_hash: state_root_hash,
```

This applies to most changes within EE - there's no need to use `state_root_hash: state_root`.

File: node/src/components/api_server/rpcs/chain.rs [hunk 3]
```diff
@@ -71,32 +71,32 @@ impl RpcWithOptionalParamsExt for GetBlock {
...
     pub api_version: Version,
-    /// Hex-encoded global state hash.
-    pub global_state_hash: Option<Digest>,
+    /// Hex-encoded hash of the state root.
+    pub state_root_hash: Option<Digest>,
 }
 
-/// "chain_get_global_state_hash" RPC.
-pub struct GetGlobalStateHash {}
+/// "chain_get_state_hash" RPC.
```
Comment (Fraser999) [inline node/src/components/api_server/rpcs/chain.rs:90]:
```suggestion
/// "chain_get_state_root_hash" RPC.
```

File: node/src/effect/requests.rs [hunk 4]
```diff
@@ -372,8 +372,8 @@ pub enum ApiRequest<I> {
     },
     /// Query the global state at the given root hash.
     QueryGlobalState {
-        /// The global state hash.
-        global_state_hash: Digest,
+        /// The root state hash.
```
Comment (Fraser999) [inline node/src/effect/requests.rs:375]:
```suggestion
        /// The state root hash.
```

File: node/src/effect/requests.rs [hunk 5]
```diff
@@ -383,8 +383,8 @@ pub enum ApiRequest<I> {
     },
     /// Query the global state at the given root hash.
     GetBalance {
-        /// The global state hash.
-        global_state_hash: Digest,
+        /// The root state hash.
```
Comment (Fraser999) [inline node/src/effect/requests.rs:386]:
```suggestion
        /// The state root hash.
```

**Review by Fraser999 — COMMENTED** [review]
I'm not sure if "hash of the state root" is accurate - maybe "hash at the state root", or I prefer just "state root hash", since the state root _is_ the hash I think?

File: execution_engine/src/core/engine_state/genesis.rs (hunk 2 above)
Comment (goral09) [reply execution_engine/src/core/engine_state/genesis.rs]: Done. I left out the places where it was referring to gRPC structs/methods.

**Review by Fraser999 — APPROVED** [review]
My only concern now is that we've gone too far with the rename inside EE.  Mainly I'm thinking of places where `post_state_hash` or `pre_state_hash` have been renamed, since possibly the prefix helped in those contexts.

If the guys mainly responsible for EE are happy with this then I'm fine with this PR too.  I'll approve, but I'd prefer if you wait for a thumbs-up from @henrytill, @mpapierski or @EdHastingsCasperLabs before merging.

**Comment by goral09** [comment]
> My only concern now is that we've gone too far with the rename inside EE. Mainly I'm thinking of places where `post_state_hash` or `pre_state_hash` have been renamed, since possibly the prefix helped in those contexts.

I share the concern. I did try to read context every time I made the change and I only remove the prefix if it was the only `_state_root_hash` in the scope.

**Comment by goral09** [comment]
Naming is a bit inconsistent, some place it's `parent_state_hash`, sometimes it's `state_hash`. Should we agree on unified naming? i.e. wherever there's no `pre_`/`parent_`/`post_` prefix I will use `state_root_hash` and change `parent_` to `pre_`?

**Comment by henrytill** [comment]
> Naming is a bit inconsistent, some place it's `parent_state_hash`, sometimes it's `state_hash`. Should we agree on unified naming? i.e. wherever there's no `pre_`/`parent_`/`post_` prefix I will use `state_root_hash` and change `parent_` to `pre_`?

Seems like that's scope creep, yah?  The ticket title is: 
> Rename `global_state_hash` to `root_hash` or `state_hash` in client facing APIs.

You are welcome to drop a ticket into the EE backlog for addressing the inconsistency.

**Review by henrytill — APPROVED** [review]

**Pull request merged**